classes/script_metadata.php

```diff
@@ -206,22 +206,35 @@ public static function get_redactable_param_names(): array {
 
     /**
      * Gets the name of the script.
-     *
+     * @param string $scriptpath
      * @return string the request path for this profile.
      */
-    public static function get_request(): string {
-        global $SCRIPT, $ME, $CFG;
+    public static function get_normalised_relative_script_path(string $scriptpath = null): string {
+        global $SCRIPT;
 
-        if (!isset($ME)) {
-            // If set, it will trim off the leading '/' to normalise web & cli requests.
-            $request = isset($SCRIPT) ? ltrim($SCRIPT, '/') : self::REQUEST_UNKNOWN;
-            return $request;
+        if (!isset($scriptpath)) {
+
+            if (!isset($SCRIPT)) {
+                return self::REQUEST_UNKNOWN;
+            }
+
+            $scriptpath = $SCRIPT;
+        } else {
+            // The $SCRIPT global has a leading slash, no consecutive slashes, and appends 'index.php' if there is a trailing slash.
+            // This else block normalises $scriptpath to match the format of $SCRIPT.
+
+            $scriptpath = '/' . $scriptpath;
+            $scriptpath = preg_replace('/\/+/', '/', $scriptpath);
+            $scriptpath = preg_replace('/\/$/', '/index.php', $scriptpath, 1);
+        }
+
+        $scriptpath = preg_replace('/\/index\.php$/', '', $scriptpath, 1);
+
+        if ($scriptpath === '') {
```

Comment:
also should get_groupby_value() be touched?

Reply:
The name could use another underscore and there is a TODO in `SCRIPT_NAMES_FOR_GROUP_REFINING` but otherwise it looks fine. Did you want me to change the name?